Formatter/Pool.php

```diff
@@ -136,4 +172,17 @@ public function getDefaultFormatter()
 
         return $this->defaultFormatter;
     }
+
+    /**
+     * Falls back to a NullLogger so that we do not have to check for the logger
+     * before calling it.
+     */
+    private function getLogger()
+    {
+        if ($this->logger) {
+            return $this->logger;
+        }
+
+        return new NullLogger();
```

Comment:
Hm, not sure about lazy initialization. I tend to avoid it. Also, you didn't spare anything, the same condition is still there. I would probably set a null logger in the constructor by default.